### PR TITLE
0.8 shady api

### PR DESCRIPTION
**Pull request by sorvell**
Adds node mutation apis to Polymer.dom: `setAttribute`, `removeAttribute`, `classList`. These methods automatically provoke distribution if necessary. Also adds redistribution support to `distributeContent`.

### CODE REVIEW

File: src/lib/dom-api.html [hunk 1]
```diff
@@ -28,6 +29,7 @@
       if (this.patch) {
         this.patch();
       }
+      this.classList = new DomApi.ClassList(this);
```
Comment (kevinpschaaf) [inline src/lib/dom-api.html]: Use getter and create lazily.

File: src/lib/dom-api.html [hunk 2]
```diff
@@ -365,10 +367,50 @@
...
+      },
+
+      removeAttribute: function(name) {
+        this.node.removeAttribute(name);
+        this._distributeParent();
+      },
+
+      _distributeParent: function() {
+        if (this.parentNode && this.parentNode.shadyRoot) {
+          this.parentNode.distributeContent();
```
Comment (kevinpschaaf) [inline src/lib/dom-api.html]: Should just call `_lazyDistribute` on the light parent.

File: src/mini/shady.html [hunk 3]
```diff
@@ -379,6 +394,35 @@
     function getComposedParent(node) {
       return node.__patched ? node._composedParent : node.parentNode;
     }
+
+    // return this host's like of redistributing hosts.
+    function getRedistributingHosts(host) {
```
Comment (kevinpschaaf) [inline src/mini/shady.html]: Can this just return the top one?

**Comment by sorvell** [comment]
@kevinpschaaf PTAL

**Comment by kevinpschaaf** [comment]
Reviewed.

**Pull request merged**